pkg/api/v1alpha1/cluster.go

```diff
@@ -872,15 +872,10 @@ func validateCloudStackK8sVersion(cluster *Cluster) error {
 
 // ValidateCloudStackK8sVersion validates version is supported by CAPC.
 func ValidateCloudStackK8sVersion(version KubernetesVersion) error {
```

Comment:
Can't we remove this provider specific validation, if we support all kubernetes version going forward for Cloudstack ?

Reply:
This validation still checks if the kubeVersion semver is valid, but I think we're good on removing this since none of the other providers have this check